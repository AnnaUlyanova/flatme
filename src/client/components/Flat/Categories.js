import React from 'react'
import { Link } from 'react-router'

import { Navbar } from 'react-bootstrap'

const Categories = props => {
  const id = props.id
  return (
    <Navbar className='categories-bar'>
      <div>
        <img src='/images/phone.svg' width='25px' />
        <Link activeClassName={'active'} to={`/flat/${id}`}> Contacts</Link>
      </div>
      <div>
        <img src='/images/interface.svg' width='25px' />
        <Link activeClassName={'active'} to={`/flat/${id}/documents`}> Documents</Link>
      </div>
      <div>
        <img src='/images/interface-1.svg' width='25px' />
        <Link activeClassName={'active'} to={`/flat/${id}/notes`}> Notes</Link>
      </div>
    </Navbar>
  )
}

export default Categories
