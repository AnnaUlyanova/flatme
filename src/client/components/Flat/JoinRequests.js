import React, { PropTypes } from 'react'

import JoinRequest from './JoinRequest'

const JoinRequests = props => {
  return (
    <div>
      <h2>Join Requests</h2>
      <ul>
        {props.requests.map((request, index) => {
          return <JoinRequest key={index} request={request} />
        })}
      </ul>
    </div>
  )
}

JoinRequests.propTypes = {
  requests: PropTypes.arrayOf(PropTypes.object).isRequired
}

export default JoinRequests
