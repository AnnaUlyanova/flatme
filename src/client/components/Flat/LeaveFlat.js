import React from 'react'
import { Button } from 'react-bootstrap'

const LeaveFlat = props => {
  return (
    <div>
      <Button onClick={props.onClick}>Leave this flat</Button>
    </div>
  )
}

export default LeaveFlat
