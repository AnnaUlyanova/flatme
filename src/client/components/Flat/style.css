/* Flat Dashboard page
–––––––––––––––––––––––––––––––––––––––––––––––––– */
.Flat .categories-bar {
  margin-top: 20px;
  padding-top: 20px;
  background-color: rgba(255,255,255,0.9);
  border: 2px solid lightseagreen;
}

.Flat .categories-bar div {
  padding-bottom: 10px;
  font-size: 1.2em;
}

.Flat .categories-bar div a {
  color: grey;
  text-align: center;
  text-decoration: none;
}

.Flat .categories-bar div a:hover {
  text-decoration: none;
  color: lightseagreen;
}

.Flat .categories-bar div .active {
  color: lightseagreen;
}
