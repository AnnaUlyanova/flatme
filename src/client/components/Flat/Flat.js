import React, { PropTypes } from 'react'
import { Row, Col } from 'react-bootstrap'

import Categories from './Categories'

const Flat = React.createClass({
  componentDidMount () {
    this.props.fetchFlat()
  },
  render () {
    const flat = this.props.flat
    const flatName = flat ? flat.flatName : null
    return (
      <div className='container-fluid'>
        <Row>
          <Col md={3}>
            <Categories id={this.props.params.id} />
          </Col>
          <Col md={9}>
            <h1>{flatName}</h1>
            {this.props.children && React.cloneElement(this.props.children, {
              flat
            })}
          </Col>
        </Row>
      </div>
    )
  }
})

Flat.propTypes = {
  fetchFlat: PropTypes.func.isRequired,
  flat: PropTypes.shape({
    flatName: PropTypes.string.isRequired
  })
}

export default Flat
