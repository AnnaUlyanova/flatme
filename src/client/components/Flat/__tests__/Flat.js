/* eslint-env jest */

import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import React from 'react'

import Flat from '..'

test('Flat renders correctly', () => {
  const fetchFlat = () => {}
  const wrapper = shallow(<Flat fetchFlat={fetchFlat} />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})
