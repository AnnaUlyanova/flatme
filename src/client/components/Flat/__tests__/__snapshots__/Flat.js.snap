exports[`test Flat renders correctly 1`] = `
<div>
  <h1 />
  <h2>
    Flatmates
  </h2>
  <ul />
</div>
`;
